src/System.Net.Security/tests/FunctionalTests/SslStreamAllowRenegotiationTests.cs

```diff
@@ -52,11 +52,8 @@ public class SslStreamAllowRenegotiationTests
                 // Initiate Read operation, that results in starting renegotiation as per server response to the above request.
                 int bytesRead = await ssl.ReadAsync(message, 0, message.Length);
 
-                // SslStream.ReadAsync should return -1 bytes, indicating renegotiation operation is in progress.
-                Assert.Equal(-1, bytesRead);
-
-                // Do another Read, to get the HTTP response from the server, after successful renegotiation.
-                bytesRead = await ssl.ReadAsync(message, 0, message.Length);
+                // There's no good way to ensure renegotiation happened in the test.
```

Comment:
Is there anything we could do by putting a custom verification stream between the network stream and the ssl stream, to see what gets read/written and validate it that way?